src/services/server/ServerService.ts

```diff
@@ -39,6 +43,26 @@ export class ServerService {
     });
   }
 
+  getServer(server: IServerAddress): ng.IPromise<IServer> {
+    let deferred: ng.IDeferred<IServer> = this.$q.defer<IServer>();
+    this.dmrService.readAttribute({
+      address: this.generateAddress(server),
+      name: "server-state"
+    }).then(serverState =>{
+      if (serverState === "STOPPED") {
+        deferred.resolve(new Server(server, serverState, ''));
```

Comment:
Should we not just reject the promise at this point with an error message? It does not make sense to me to create a server object when there isn't one available. In the controller we can then just handle a rejected promise and maybe even display the rejected message?